src/backend/distributed/commands/table.c

```diff
@@ -939,6 +939,32 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					Relation rel = RelationIdGetRelation(leftRelationId);
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
+														  RelationGetNamespace(rel),
+														  NULL, NULL, true, true);
+					RelationClose(rel);
+
+					Assert(GetMemoryChunkContext(constraint->conname) ==
```

Comment:
> I change the original command intentionally. The idea is that we should run the same command, with the name we explicitly created, across all the tables on the coordinator as well as the workers. Hence I change the original command so that the standard utility function uses the new name.

After some more thought, I realize that you and Marco make a good point. Let's change the original command. But let's document this well with comments.

In that case, we don't need to touch newCmd at all. However, I realize you did that because of the following 
```C
if (deparseAT)
	{
		newStmt->cmds = list_make1(newCmd);
		sqlForTaskList = DeparseTreeNode((Node *) newStmt);
	}
```
We should refactor this part of the code, because it was specifically written for cases where you DON'T want to modify the original command. We might need another flag.